demos/spawn/README.md

```diff
@@ -0,0 +1,40 @@
+# Spawn
+
+In this demo we are going to show the mechanism that a thread can spawn other threads.
+
+To illustrate this demo, we need to introduce the following:
+
+* `add_named_thread` (in `src/user/thread.c`)
+* `thread_name` (in `src/user/thread.c`)
+
+## `add_named_thread`
+
+`add_named_thread` uses the `name` and `worker` we provided to create a thread and initializes all its arguments to `0`.
+
+If the creation successes, the function returns the index of the thread as the return value. If the creation fails, the return value will be set to `INVALID_THREAD`.
+
+After the creation, the thread starts to wait until it is yielded to.
+
+## `thread_name`
+
+In AMT, we store the name of a thread in a `Thread` structure. 
+
+When we call this function, it uses the `get_thread_property` syscall to copy the name from the kernel, then stores it into the location pointed to by `name`. 
```

Comment:
For the purposes of the demo I think this is fine.

I did just add all that C++ weirdness but it's not worth going into here, syscall name is enough.